VSConfigFinder/Utilities.cs

```diff
@@ -70,5 +70,29 @@ private static string CreateCommandLineOutput(VSConfig finalConfig)
 
             return output.ToString();
         }
+
+        public static string[] ReadComponents(CommandLineOptions options)
+        {
+            var fileSystem = new FileSystem();
+            var pathsToVsConfigs = fileSystem.GetFileSystemEntries(options.FolderPath, ConfigExtension, true);
+
+            HashSet<string> componentsSet = new HashSet<string>();
+            var serializerOptions = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
+
+            foreach (var path in pathsToVsConfigs)
+            {
+                var stream = fileSystem.OpenFile(path);
+                var components = JsonSerializer.Deserialize<VSConfig>(stream, serializerOptions).Components;
```

Comment:
Should we wrap this around so that we can deal with exceptions? + Do a null check on the deserialized output

Reply:
Yea null check is done below, and both OpenFile and Deserialize can throw exception, I will write try catch block. When we catch it, should we simply Console.WriteLine to let user know what happened?

Reply:
For example, `JsonSerializer.Deserialize<T>` returns an `object?` which means that the output could be null. If we don't do a null check here in any way, doing a `.Components` after that could throw.

Upon catch, I'd say we write a log through `Console.WriteLine` to signal the user that it failed & why it failed.